src/EFCore.PG/Query/Expressions/Internal/FullTextSearchExpression.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Linq;
+using System.Linq.Expressions;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Query.Sql.Internal;
+using Microsoft.EntityFrameworkCore.Utilities;
+using NpgsqlTypes;
+
+namespace Microsoft.EntityFrameworkCore.Query.Expressions.Internal
+{
+    public class FullTextSearchExpression : Expression
+    {
+        private FullTextSearchExpression(
+            string binaryOperator,
+            [NotNull] Expression left,
+            [CanBeNull] Expression right,
+            [NotNull] Type type)
+        {
+            Check.NotEmpty(binaryOperator, nameof(binaryOperator));
+            Check.NotNull(left, nameof(left));
+            Check.NotNull(type, nameof(type));
+
+            Operator = binaryOperator;
+            Left = left;
+            Right = right;
+            Type = type;
+        }
+
+        public string Operator { get; }
+        public Expression Left { get; }
+        public Expression Right { get; }
+        public override Type Type { get; }
+        public override ExpressionType NodeType => ExpressionType.Extension;
+
+        protected override Expression Accept(ExpressionVisitor visitor)
+        {
+            Check.NotNull(visitor, nameof(visitor));
+
+            return visitor is NpgsqlQuerySqlGenerator npgsqlGenerator
+                ? npgsqlGenerator.VisitFullTextSearch(this)
+                : base.Accept(visitor);
+        }
+
+        protected override Expression VisitChildren(ExpressionVisitor visitor)
+        {
+            var newLeft = visitor.Visit(Left);
+            var newRight = Right != null ? visitor.Visit(Right) : null;
+
+            return newLeft != Left
+                   || newRight != Right
+                ? new FullTextSearchExpression(Operator, newLeft, newRight, Type)
+                : this;
+        }
+
+        public override string ToString() => Right != null
+            ? $"{Left} {Operator} {Right}"
+            : $"{Operator} {Left}";
+
+        public static FullTextSearchExpression TsQueryAnd([NotNull] Expression left, [NotNull] Expression right)
```

Comment:
I'm not sure this actually have much value... If/when you you refactor the expression as I suggested, creating a "contains" expression would be a matter of instantiating a FullTextSearchBinaryExpression and passing an enum to the constructor (`NpgsqlFullTextOperation.Contains` or something similar).